profiler/upload_test.go

```diff
@@ -48,57 +49,83 @@ func TestTryUpload(t *testing.T) {
 	defer func(cid string) { containerID = cid }(containerID)
 	containerID = ""
 
-	srv, srvURL, waiter := makeTestServer(t, 200)
-	defer srv.Close()
-	p, err := unstartedProfiler(
-		WithAgentAddr(srvURL.Host),
-		WithService("my-service"),
-		WithEnv("my-env"),
-		WithTags("tag1:1", "tag2:2"),
-	)
-	require.NoError(t, err)
-	err = p.doRequest(testBatch)
-	require.NoError(t, err)
-	header, fields, tags := waiter()
-
-	assert := assert.New(t)
-	assert.Empty(header.Get("Datadog-Container-ID"))
-	assert.ElementsMatch([]string{
-		"host:my-host",
-		"runtime:go",
-		"service:my-service",
-		"env:my-env",
-		"tag1:1",
-		"tag2:2",
-		fmt.Sprintf("pid:%d", os.Getpid()),
-		fmt.Sprintf("profiler_version:%s", version.Tag),
-		fmt.Sprintf("runtime_version:%s", strings.TrimPrefix(runtime.Version(), "go")),
-		fmt.Sprintf("runtime_compiler:%s", runtime.Compiler),
-		fmt.Sprintf("runtime_arch:%s", runtime.GOARCH),
-		fmt.Sprintf("runtime_os:%s", runtime.GOOS),
-		fmt.Sprintf("runtime-id:%s", globalconfig.RuntimeID()),
-	}, tags)
-	for k, v := range map[string]string{
-		"format":   "pprof",
-		"runtime":  "go",
-		"types[0]": "cpu",
-		"data[0]":  "my-cpu-profile",
-		"types[1]": "alloc_objects,alloc_space",
-		"data[1]":  "my-heap-profile",
-	} {
-		assert.Equal(v, fields[k], k)
+	fixtures := []struct {
+		description     string
+		startTestServer startTestServerFn
+		exopts          func(string) []Option
+	}{
+		{
+			description:     "test against httptest.Server",
```

Comment:
The parameter passed to `t.Run` should not be more than one word. In Go this is usually meant to be more of an identifier than a sentence or description of the test (like in other languages `it ...`). In our case we can have two sub-tests: one called `uds` and one called `http`.

Reply:
Strong disagree on the one-word guideline, but will duplicate the test server code and remove the abstraction, making it moot.